tron/api/resource.py

```diff
@@ -350,13 +357,45 @@ def render_GET(self, request):
             "include_node_pool",
             default=True,
         )
+
+        page = requestargs.get_integer(request, "page")
+        page_size = requestargs.get_integer(request, "page_size")
+
+        jobs = self.get_data(
+            include_job_runs,
+            include_action_runs,
+            include_action_graph,
+            include_node_pool,
+            page,
+            page_size,
+        )
+
+        if page is not None and page_size is not None:
+            total = len(self.job_collection.get_jobs())
+            if page < 1 or page_size < 1:
+                return respond(
+                    request=request,
+                    response={"error": "page and page_size must be positive integers"},
+                    code=http.BAD_REQUEST,
+                )
+            if page > math.ceil(total / page_size):
+                return respond(
+                    request=request,
+                    response={"error": "page out of range"},
+                    code=http.BAD_REQUEST,
+                )
+            response = dict(jobs=jobs, pagination=dict(page=page, page_size=page_size, total=total))
+        else:
+            response = dict(jobs=jobs)
+        return respond(request=request, response=response)
+
+    @AsyncResource.bounded
+    def render_SEARCH(self, request):
+        query = requestargs.get_string(request, "query")
```

Comment:
do we need to do any validation on the `query` data?

Reply:
Search is currently very basic we could maybe limit the length of the string 🤷🏽 we also don't use it in a way that could lead to SQL injections